be-bop-wizard/be-bop-wizard.sh

```diff
@@ -1923,7 +1965,28 @@ server {
 EOF
     else
         # Production configuration with HTTPS
-        sed "s/example.com/${domain}/g" > "$TMPDIR/be-BOP.conf" << 'EOF'
+        # Build domain lists for different server blocks
+        local all_domains="$(build_domain_list "$domain")"
+
+        # For main HTTPS server block:
+        # - If domain is www.example.com, ONLY serve www.example.com (not example.com)
+        # - If domain is example.com, serve example.com
+        # - Never include s3.* domains
+        # This prevents CORS errors by ensuring the root domain only hits the redirect block
+        if [[ "$domain" =~ ^www\. ]]; then
+            # For www domains, main server should ONLY serve the www domain
+            local main_domains="$domain"
+        else
+            # For non-www domains, include all non-s3 domains
+            local main_domains=$(echo "$all_domains" | tr ' ' '\n' | grep -v '^s3\.' | tr '\n' ' ' | sed 's/ $//')
+        fi
+
+        local s3_domain=$(echo "$all_domains" | tr ' ' '\n' | grep '^s3\.' | head -1)
+
+        sed -e "s/server_name example.com s3.example.com;/server_name ${all_domains};/" \
+            -e "s/server_name example.com;/server_name ${main_domains};/" \
+            -e "s/server_name s3.example.com;/server_name ${s3_domain};/" \
+            > "$TMPDIR/be-BOP.conf" << 'EOF'
 # The sites in this file are managed by be-bop-wizard
```

Comment:
We need to add a fingerprint to this file, similar to https://github.com/be-BOP-io-SA/be-BOP-tooling/commit/554d25b61dcb9a303cf5a7fde8acd3858389d9c6.